packages/providers-evm/src/base-evm.ts

```diff
@@ -31,13 +31,18 @@ export class BaseEVMProvider extends ProviderEventBus implements ProviderProxy {
   #chainId?: ChainId
   #address?: string
 
-  constructor(provider?: RawProvider) {
+  constructor(provider?: RawProvider | providers.Web3Provider) {
     if (!provider) throw new errors.ProviderInjectedInstanceNotFoundError()
     super()
-    this.#provider = new ethers.providers.Web3Provider(
-      wrapExternalEthProvider(provider as ethers.providers.ExternalProvider),
-      'any',
-    )
+    this.#provider =
+      provider instanceof providers.Web3Provider
+        ? provider
+        : new ethers.providers.Web3Provider(
+            wrapExternalEthProvider(
+              provider as ethers.providers.ExternalProvider,
+            ),
+            'any',
+          )
```

Comment:
try to do this, so we will be able to catch errors and wrap them into the our ones
```ts
    const rawProvider = wrapExternalEthProvider(
      provider instanceof providers.Web3Provider
        ? provider.provider
        : (provider as providers.ExternalProvider),
    )

    this.#provider = new providers.Web3Provider(rawProvider, 'any')
```

You should try to call the switch chain method on the provider and reject it for example to ensure that errors are wrapped into the our ones:

<img width="1054" alt="image" src="https://github.com/rarimo/js-sdk/assets/47495186/29857b86-9d84-4f3b-8513-e18d73a57e61">

Reply:
done